File: mek_stripe_terminal/android/src/main/kotlin/mek/stripeterminal/mappings/SimulatedOfflineModeMappings.kt
package mek.stripeterminal.mappings

import com.stripe.stripeterminal.external.models.SimulatedOfflineMode
import com.stripe.stripeterminal.external.models.SimulatedOfflineModeConfiguration
import mek.stripeterminal.api.SimulatedOfflineModeApi
import mek.stripeterminal.api.SimulatedOfflineModeConfigurationApi

fun SimulatedOfflineModeConfigurationApi.toHost(): SimulatedOfflineModeConfiguration {
    return SimulatedOfflineModeConfiguration(
        sdkOfflineMode = sdkOfflineMode.toHost(),
        readerOfflineMode = readerOfflineMode.toHost()
    )
}

fun SimulatedOfflineModeApi.toHost(): SimulatedOfflineMode {
    return when (this) {
        SimulatedOfflineModeApi.DISABLED -> SimulatedOfflineMode.DISABLED
        SimulatedOfflineModeApi.OFFLINE_IMMEDIATE -> SimulatedOfflineMode.OFFLINE_IMMEDIATE
        SimulatedOfflineModeApi.OFFLINE_TIMEOUT -> SimulatedOfflineMode.OFFLINE_TIMEOUT
        SimulatedOfflineModeApi.OFFLINE_INTERMITTENT -> SimulatedOfflineMode.OFFLINE_INTERMITTENT
    }
}

File: mek_stripe_terminal/ios/mek_stripe_terminal/Sources/mek_stripe_terminal/Mappings/SimulatedOfflineModeMappings.swift
import Foundation
import StripeTerminal

extension SimulatedOfflineModeConfigurationApi {
    func toHost() throws -> SimulatedOfflineModeConfiguration {
        return try SimulatedOfflineModeConfigurationBuilder()
            .setSdkOfflineMode(sdkOfflineMode.toHost())
            .setReaderOfflineMode(readerOfflineMode.toHost())
            .build()
    }
}

extension SimulatedOfflineModeApi {
    func toHost() -> SimulatedOfflineMode {
        switch self {
        case .disabled:
            return .disabled
        case .offlineImmediate:
            return .offlineImmediate
        case .offlineTimeout:
            return .offlineTimeout
        case .offlineIntermittent:
            return .offlineIntermittent
        }
    }
}
